core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/storage/FileResolver.scala

```diff
@@ -0,0 +1,57 @@
+package edu.uci.ics.texera.workflow.common.storage
+
+import java.nio.file.{Files, Path, Paths}
+import edu.uci.ics.amber.engine.common.storage.DatasetFileDocument
+import org.apache.commons.vfs2.FileNotFoundException
+
+import scala.util.{Failure, Success, Try}
+
+object FileResolver {
+
+  type ResolverOutput = Either[String, DatasetFileDocument]
```

Comment:
Good. In the next PR you can change this to return string only

Reply:
Sure